### PR TITLE
MWPW-122988 - [Library] Search & IDs

### PR DESCRIPTION
**Pull request by jckautzmann**
- adds a search field in the library sidekick to search for blocks 

Screenshot 2023-02-02 at 17 17 23

Features:
- the search field is only displayed when browsing for `Blocks`. Otherwise it is hidden as the number of `Placeholders` and `Icons` is very limited
- looks up blocks based on the block title (e.g. `Marquee (split, light)`)
- supports block IDs/tokens defined by authors in the `searchTags` property of the section metadata following the block
Screenshot 2023-02-02 at 17 12 20
- the feature can be viewed/tested/verified at:
https://library-search--milo--adobecom.hlx.page/tools/library?martech=off

Resolves: [MWPW-122988](https://jira.corp.adobe.com/browse/MWPW-122988)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/tools/library?martech=off
- After: https://library-search--milo--adobecom.hlx.page/tools/library?martech=off

### CODE REVIEW

**Comment by ryanmparrish** [comment]
Great addition!
Stupid question - how do I get my library to point to your branch? I've tired checking it out and looking at it locally. 
The library is only available when sidekick is used in Sharepoint, but it looks like it's not loading that branch as there is not search feature present. LMK if i'm missing something on the testing. Thanks @jckautzmann

**Comment by rgclayton** [comment]
Sample pages for tools - https://library-search--milo--adobecom.hlx.page/tools/library?martech=off

File: libs/blocks/library-config/lists/blocks.js [hunk 1]
```diff
@@ -50,10 +51,31 @@ function getTable(block, name, path) {
...
+
+function isMatchingBlock(pageBlock, query) {
+  const tagsString = getBlockTags(pageBlock);
+  if (!query || !tagsString) return false;
+  const searchTokens = query.split(' ');
+  return searchTokens.every((token) => tagsString.includes(token));
+}
+
+export default async function loadBlocks(blocks, list, query) {
+  list.innerHTML = '';
```
Comment (rgclayton) [inline libs/blocks/library-config/lists/blocks.js:72]: `innerHTML` can be a security risk. It's better to use `textContent`

```suggestion
  list.textContent = '';
```

File: libs/blocks/library-config/library-config.css [hunk 2]
```diff
@@ -57,6 +61,51 @@
...
+  margin-left: 5px;
+  top: 10px;
+}
+
+input.sk-library-search-input {
+  width: 100%;
+  padding-left: 30px;
+  background: rgb(0 0 0 / 35%);
+  color: #FFF;
+  font-family: 'Adobe Clean', adobe-clean, 'Trebuchet MS', sans-serif;
```
[the remark this answers is not in the document]
Comment (jckautzmann) [reply libs/blocks/library-config/library-config.css]: good catch! Addressed with https://github.com/adobecom/milo/pull/437/commits/a1bfd9f0229a15b732f7ea5a8669c954a701bdb8

File: libs/blocks/library-config/lists/blocks.js (hunk 1 above)
Comment (jckautzmann) [reply libs/blocks/library-config/lists/blocks.js]: I changed it in https://github.com/adobecom/milo/pull/437/commits/a1bfd9f0229a15b732f7ea5a8669c954a701bdb8

**Review by rgclayton — APPROVED** [review]
This is looking good. 

I made a comment in this Jira ticket related to creating a new ticket or subtask  to populate the `searchTags` property for all blocks.

**Review by chrischrischris — APPROVED** [review]

**Pull request merged**